src/Compiler/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/DefaultDocumentWriter.cs

```diff
@@ -33,12 +35,15 @@ public override RazorCSharpDocument WriteDocument(RazorCodeDocument codeDocument
             throw new ArgumentNullException(nameof(documentNode));
         }
 
+        using var _ = ArrayBuilderPool<SourceMapping>.GetPooledObject(out var sourceMappingsBuilder);
+
         var context = new DefaultCodeRenderingContext(
             new CodeWriter(Environment.NewLine, _options),
             _codeTarget.CreateNodeWriter(),
             codeDocument,
             documentNode,
-            _options);
+            _options,
+            sourceMappingsBuilder);
```

Comment:
What is the lifetime of `DefaultCodeRenderingContext` like? When created, is it always immediately thrown away? If so, should it be made `IDisposable` and allow it to own the `ImmutableArray<SourceMapping>.Builder` rather than passing one in?

Reply:
Outside of tests, the lifetime is very short, yet. Will update.